botorch/test_functions/synthetic.py

```diff
@@ -802,7 +805,57 @@ def evaluate_true(self, X: Tensor) -> Tensor:
 #  ------------ Constrained synthetic test functions ----------- #
 
 
-class ConstrainedGramacy(ConstrainedBaseTestProblem, SyntheticTestFunction):
+class ConstrainedSyntheticTestFunction(
+    ConstrainedBaseTestProblem, SyntheticTestFunction
+):
+    r"""Base class for constrained synthetic test functions."""
+
+    def __init__(
+        self,
+        noise_std: Union[None, float, List[float]] = None,
+        constraint_noise_std: Union[None, float, List[float]] = None,
+        negate: bool = False,
+        bounds: Optional[List[Tuple[float, float]]] = None,
+    ) -> None:
+        r"""
+        Args:
+            noise_std: Standard deviation of the observation noise. If a list is
+                provided, specifies separate noise standard deviations for each
+                objective in a multiobjective problem.
+            constraint_noise_std: Standard deviation of the constraint noise.
+                If a list is provided, specifies separate noise standard
+                deviations for each constraint.
+            negate: If True, negate the function.
+            bounds: Custom bounds for the function specified as (lower, upper) pairs.
+        """
+        self.setup_constraint_noise(constraint_noise_std)
+        SyntheticTestFunction.__init__(
+            self, noise_std=noise_std, negate=negate, bounds=bounds
+        )
+
+    def setup_constraint_noise(self, constraint_noise_std):
+        """
+        Validates that constraint_noise_std has length equal to
+        the number of constraints, if given as a list
+
+        Args:
+            constraint_noise_std: Standard deviation of the constraint noise.
+                If a list is provided, specifies separate noise standard
+                deviations for each constraint.
+        """
+        if (
+            isinstance(constraint_noise_std, list)
+            and len(constraint_noise_std) != self.num_constraints
+        ):
+            raise InputDataError(
+                f"If specified as a list, length of constraint_noise_std "
+                f"({len(constraint_noise_std)}) must match the "
+                f"number of constraints ({self.num_constraints})"
+            )
+        self.constraint_noise_std = constraint_noise_std
```

Comment:
Minor style suggestion: I prefer these kinds of helpers to primarily validate but do assignments that change class attributes in the constructor itself. Nothing wrong with your approach, but doing this makes it more clear where things are happening / the code easier to read IMO. 

```suggestion
        self.constraint_noise_std = self._validate_constraint_noise(constraint_noise_std)
        SyntheticTestFunction.__init__(
            self, noise_std=noise_std, negate=negate, bounds=bounds
        )

    def _validate_constraint_noise(self, constraint_noise_std) -> Union[None, float, List[float]]:
        """
        Validates that constraint_noise_std has length equal to
        the number of constraints, if given as a list

        Args:
            constraint_noise_std: Standard deviation of the constraint noise.
                If a list is provided, specifies separate noise standard
                deviations for each constraint.
        """
        if (
            isinstance(constraint_noise_std, list)
            and len(constraint_noise_std) != self.num_constraints
        ):
            raise InputDataError(
                "If specified as a list, length of constraint_noise_std "
                f"({len(constraint_noise_std)}) must match the "
                f"number of constraints ({self.num_constraints})"
            )
        return constraint_noise_std
```